client/app/dbaas/logs/detail/streams/archives/streams-archives.controller.js

```diff
@@ -0,0 +1,254 @@
+    }
+
+    $onInit () {
+        this.archiveIds.load();
+        this.stream.load();
+        this.notifications = [];
+    }
+
+    $destroy () {
+        this._stopRetrievalDelayUpdate();
+    }
+
+    /**
+     * Gets the notificaiton index for an archive if available
+     *
+     * @param {any} archive
+     * @returns the notification index. Returns -1 if not found
+     * @memberof LogsStreamsArchivesHomeCtrl
+     */
+    _getNotificationIndex (archive) {
+        return this.notifications.reduce((matchedIndex, currentNotification, currentIndex) => currentNotification.archive.filename === archive.filename ? currentIndex : matchedIndex, -1);
+    }
+
+    /**
+     * initializes the archivesIDs and current stream
+     *
+     * @memberof LogsStreamsArchivesHomeCtrl
+     */
+    _initLoaders () {
+        this.archiveIds = this.ControllerHelper.request.getArrayLoader({
+            loaderFunction: () => this.LogsStreamsArchivesService.getArchiveIds(this.serviceName, this.streamId)
+        });
+        this.stream = this.ControllerHelper.request.getHashLoader({
+            loaderFunction: () => this.LogsStreamsService.getStream(this.serviceName, this.streamId)
+        });
+    }
+
+    /**
+     * Updates the list of archive with the latest information of the archive
+     *
+     * @param {any} archiveId
+     * @returns promise which will be resolve with the reloaded archive
+     * @memberof LogsStreamsArchivesHomeCtrl
+     */
+    _reloadArchiveDetail (archiveId) {
+        this.archiveReload = this.ControllerHelper.request.getArrayLoader({
+            loaderFunction: () => this.LogsStreamsArchivesService.getArchives(
+                this.serviceName,
+                this.streamId,
+                [archiveId]
+            )
+        });
+
+        return this.archiveReload.load()
+            .then(archives => {
+                const archive = archives[0];
+                this.archives.data.forEach((archiveItem, archiveIndex) => {
+                    if (archiveItem.archiveId === archive.archiveId) {
+                        this.archives.data[archiveIndex] = archive;
+                    }
+                });
+                return archive;
+            });
+    }
+
+    /**
+     * Removes the notification for an archive if available
+     *
+     * @param {any} archive
+     * @returns Returns the removed notification
+     * @memberof LogsStreamsArchivesHomeCtrl
+     */
+    _removeNotification (archive) {
+        const notificationIndex = this._getNotificationIndex(archive);
+        return this.notifications.splice(notificationIndex, notificationIndex >= 0 ? 1 : 0);
+    }
+
+    /**
+     * Starts an interval that runs every second, that updates
+     * the remaining unsealing time for all archives being unsealed
+     *
+     * @memberof LogsStreamsArchivesHomeCtrl
+     */
+    _startRetrievalDelayUpdate () {
+        this.retrievalDelayUpdater = this.$interval(this._updateRetrievalDelay.bind(this), 1000);
```

Comment:
You can use CloudPoll for that and if you don't wanna use CloudPoll: this.$interval(() => this._updateRetrievalDelay(), 1000);

Reply:
The polling has to be stopped and restarted multiple times and that's the reason why I feel more comfortable using a $interval instead of CloudPoll. I have removed the bind and used $interval, the way you have mentioned. Thanks